Umbrella checklist kickoff!

aainscow opened the pull request:
This is a start at the Umbrella Kickoff.  

I have implemented the parts that are necessary for the client versioning I need to add. 

Tracker: https://tracker.ceph.com/issues/74141
<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)

You must only issue one Jenkins command per-comment. Jenkins does not understand
comments with more than one command.
</details>

aainscow committed:
CMakeLists.txt: update VERSION

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
librbd: bump version

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
ceph_release: update to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
*: add constants and release names

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
doc/dev/release-checklist: add backport-create-issue

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
cephadm,ceph-volume: update to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
include/ceph_features: add SERVER_UMBRELLA feature bit

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
doc: update compatset for umbrella

The unknown bit shown in commit was picked as being "far in the future".
We have now used it and now need to move the bit one release at a time.

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
qa/standalone/mon/misc: update for umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
mon/MgrMonitor: update for umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
common/options/global.yaml.in: update for umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
mon/OSDMonitor: update to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
qa/workunits/cephtool/test: update to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
osd/OSDMap: update to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
tools/monmaptool: bump new cluster version to X

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
qa: update require-osd-release to umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
doc: reset for umbrella

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

batrick commented on src/include/ceph_features.h:
```diff
@@ -162,7 +162,7 @@ DEFINE_CEPH_FEATURE(49, 2, SERVER_SQUID);
 DEFINE_CEPH_FEATURE_RETIRED(50, 1, MON_METADATA, MIMIC, OCTOPUS)
 DEFINE_CEPH_FEATURE(50, 2, SERVER_TENTACLE);
 DEFINE_CEPH_FEATURE_RETIRED(51, 1, OSD_BITWISE_HOBJ_SORT, MIMIC, OCTOPUS)
```
I think this can be removed. cc @rzarzynski

batrick reviewed (commented):
Thanks for starting this!

bill-scales reviewed (commented):
Changes so far look good.

src/common/ceph_strings.cc:: ceph_release_features (line 171) will need updating for umbrella because we are changing the client->OSD protocol for both direct reads and pool migration

aainscow commented:
@bill-scales 
> Changes so far look good.
> 
> src/common/ceph_strings.cc:: ceph_release_features (line 171) will need updating for umbrella because we are changing the client->OSD protocol for both direct reads and pool migration

As in this?

```
Index: src/common/ceph_strings.cc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/common/ceph_strings.cc b/src/common/ceph_strings.cc
--- a/src/common/ceph_strings.cc	(revision 3fa00db04dda621b52c55aa7dfc883e9659d5494)
+++ b/src/common/ceph_strings.cc	(date 1765271478614)
@@ -166,7 +166,11 @@
 		return req;
 
 	req |= CEPH_FEATUREMASK_CRUSH_MSR;
-	if (r <= CEPH_RELEASE_SQUID)
+	if (r <= CEPH_RELEASE_TENTACLE)
+		return req;
+
+	req |= CEPH_FEATUREMASK_SERVER_UMBRELLA;
+	if (r <= CEPH_RELEASE_UMBRELLA)
 		return req;
 
 	return req;
```

Do we need new feature bits for DIRECT READS and POOL requests?

aainscow committed:
doc: Confirm DEPRECATED checklist item and clean up pg_scrubber

No DEPRECATED features.

Remove some code labeled as "Remove in Umbrella"

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

sseshasa commented:
> Should we be removing the else branch here @sseshasa ?
> 
The else branch can be retained and the comment updated to mention  "Umbrella".

Commit:
qa Upgrade to Umbrella

All the qa tasks in the umbrella release checklist

Signed-off-by: Chris Harris <harriscr@uk.ibm.com>
Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

# Conflicts:
#	qa/suites/upgrade/telemetry-upgrade/tentacle-x/1-tasks.yaml
#	qa/workunits/test_telemetry_tentacle.sh
#	qa/workunits/test_telemetry_tentacle_x.sh

aainscow committed:
osd: Update remove after squid to remove after umbrella comment

Following consulting with author, we don't yet want to remove.

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
osd: Remove octopus-depending code in Fast EC

We are leaving alone legacy EC

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

NitzanMordhai commented on doc/dev/release-checklists.rst:
```diff
@@ -87,40 +87,40 @@ OSDMap
 - [x] create qa/releases/X.yaml
-- [x] create qa/suites/rados/thrash-old-clients/1-install/(X-1).yaml
+- [ ] create qa/suites/rados/thrash-old-clients/1-install/(X-1).yaml
 - [x] update qa/suites/rados/encoder/1-task.yaml to remove (X-3) and add X
```
Lets change that too, so we won't miss it

Commit:
qa: remove reef code from fs

Remove the reef code from the fs upgrade suites as we no longer want to test X-3 releases

Signed-off-by: Chris Harris <harriscr@uk.ibm.com>

Commit:
qa: add X-1 release (tentacle) to fs upgrade suite

Add files for the tentacle release (X-1) to the qa/suites/fs/upgrade suites

Signed-off-by: Chris Harris <harriscr@uk.ibm.com>

rzarzynski commented on doc/dev/release-checklists.rst:
```diff
@@ -16,7 +16,7 @@ Versions and tags
 
 - [x] Update CMakeLists.txt VERSION (right at the top to X.0.0)
-- [ ] Update src/librbd/CMakeLists.txt VERSION (librbd target at the bottom to 1.X.0)
+- [x] Update src/librbd/CMakeLists.txt VERSION (librbd target at the bottom to 1.X.0)
```
ACK.

rzarzynski commented on src/mon/mon_types.h:
```diff
@@ -615,6 +618,9 @@ namespace ceph {
 static inline ceph_release_t infer_ceph_release_from_mon_features(mon_feature_t f)
 {
+  if (f.contains_all(ceph::features::mon::FEATURE_UMBRELLA)) {
+    return ceph_release_t::tentacle;
```
`umbrella`?

aainscow replied:
WIll Fix

aainscow replied:
Fixed

rzarzynski commented on src/include/ceph_features.h:
```diff
@@ -162,9 +162,8 @@ DEFINE_CEPH_FEATURE(49, 2, SERVER_SQUID);
-// available
 DEFINE_CEPH_FEATURE_RETIRED(52, 1, OSD_PROXY_WRITE_FEATURES, MIMIC, OCTOPUS)
-// available
+DEFINE_CEPH_FEATURE(52, 2, SERVER_UMBRELLA);
```
The bit selection looks fine.
Perhaps we should consider cleaning the `// available` comments but it's an old debt.

rzarzynski commented on doc/dev/release-checklists.rst:
```diff
@@ -61,11 +61,11 @@ Feature bits
+- [x] mon/Monitor.cc (include in `get_supported_features()`)
+- [x] mon/Monitor.cc (`apply_monmap_to_compatset_features()`)
+- [x] mon/Monitor.cc (`calc_quorum_requirements()`)
+- [x] test/cli/monmaptool/feature-set-unset-list.t (`supported`, `persistent`)
```
```
$ git show --name-only 3c62515cca84c62ac440e42d62528c16e4936603
commit 3c62515cca84c62ac440e42d62528c16e4936603
Author: Alex Ainscow <aainscow@uk.ibm.com>
Date:   Thu Jan 9 23:44:25 2025 -0500

    doc: update compatset for tentacle
    
    Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

doc/dev/release-checklists.rst
src/mon/Monitor.cc
src/mon/Monitor.h
src/test/cli/monmaptool/feature-set-unset-list.t
```

rzarzynski commented on src/mon/OSDMonitor.cc:
TOCHECK: have we missed bumping this up?

rzarzynski commented on doc/dev/release-checklists.rst:
```diff
@@ -108,7 +108,7 @@ QA suite
 - [ ] update qa/fs/upgrade/ to remove (X-3) and add (X-1); check with fs team to confirm / help
 - [ ] update qa/ upgrade suites require-osd-release calls to tentacle
-- [ ] create qa/releases/X.yaml
+- [x] create qa/releases/X.yaml
```
Oops, I can't fine the `umbrella.yaml` in the `qa/releases` dir.

```
$ git show --name-only bfc18bbf449344cb774d20e508341ae428c00c39
commit bfc18bbf449344cb774d20e508341ae428c00c39
Author: Alex Ainscow <aainscow@uk.ibm.com>
Date:   Mon Feb 17 18:34:56 2025 +0000

    qa: update require-osd-release to tentacle
    
    Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

doc/dev/release-checklists.rst
qa/suites/fs/upgrade/featureful_client/old_client/tasks/2-upgrade.yaml
qa/suites/fs/upgrade/featureful_client/upgraded_client/tasks/2-upgrade.yaml
qa/suites/fs/upgrade/nofs/tasks/1-upgrade.yaml
qa/suites/upgrade/telemetry-upgrade/reef-x/1-tasks.yaml
qa/suites/upgrade/telemetry-upgrade/squid-x/1-tasks.yaml
```

(IMPORTANT)

rzarzynski reviewed (commented):
1st round.

aainscow committed:
docs: reorder mon and mds for future new releases

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
qa: qa tasks from the release checklist for Umbrella

All the qa tasks in the umbrella release checklist.
This includes removing X-3 releases and adding X-1 releases
(where appropriate).

Also updated release-checklist.rst

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
include: Remove incorrect "available" comment for feature bit. (cleanup)

This flag was not available, so I have deleted it.

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow commented on src/test/cli/monmaptool/feature-set-unset-list.t:
```diff
@@ -12,21 +12,21 @@
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512),squid(1024),tentacle(2048),nvmeof_beacon_diff(4294967296)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512),squid(1024),tentacle(2048),nvmeof_beacon_diff(4294967296)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512),squid(1024),tentacle(2048),nvmeof_beacon_diff(4294967296),umbrella(4096)]
```
I don't understand. Can you rephrase the request?

aainscow commented on src/common/options/global.yaml.in:
```diff
@@ -2037,19 +2037,19 @@ options:
   see_also:
   - mon_debug_dump_transactions
   with_legacy: true
-- name: mon_debug_no_require_squid
```
I have checked - this does seem to have been done. 

Please resolve if you agree.

aainscow commented:
/config check ok

adk3798 reviewed (approved):
minor cephadm changes (just the default release `tentacle` -> `umbrella` unless I'm missing something) LGTM. Once we actually branch off and have umbrella builds we'll also need to change the default image itself (in the same file) to point to that instead of main, but I don't think that would work at this stage in the process

yaarith commented on doc/dev/release-checklists.rst:
```diff
@@ -74,7 +74,7 @@ Mon
 - [ ] qa/standalone/mon/misc adjust `TEST_mon_features` (add X cases and adjust `--mon-debug-no-require-X`)
 - [x] qa/standalone/mon/misc bump up `jqfilter='.monmap.features.persistent | length == N'` to `N+1`
-- [ ] mon/MgrMonitor.cc adjust `always_on_modules`
+- [x] mon/MgrMonitor.cc adjust `always_on_modules`
```
@rzarzynski thanks, I just replied (https://github.com/ceph/ceph/pull/66524/changes#r2775271623).

rzarzynski commented:
The failing ceph dashboard check is because of:

```
Feb 26 14:30:07 ceph-node-00 cloud-init[994]: Ceph version: ceph version 20.3.0-5603-g237566a0 (237566a068813544160c5a938c95ff00f7ace50c) tentacle (dev)
Feb 26 14:30:07 ceph-node-00 cloud-init[994]: Error: Container release tentacle != cephadm release umbrella; please use matching version of cephadm (pass --allow-mismatched-release to continue anyway)
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         ***************
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         Cephadm hit an issue during cluster installation. Current cluster files will be deleted automatically.
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         To disable this behaviour you can pass the --no-cleanup-on-failure flag. In case of any previous
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         broken installation, users must use the following command to completely delete the broken cluster:
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         > cephadm rm-cluster --force --zap-osds --fsid <fsid>
Feb 26 14:30:07 ceph-node-00 cloud-init[994]:         for more information
```

Looks we need to merge the `ceph-build` PR (https://github.com/ceph/ceph-build/pull/2499) first, after resolving the @djgalloway comments there.

aainscow commented on doc/dev/release-checklists.rst:
```diff
@@ -31,12 +31,12 @@ Define release names and constants
 
 Make sure X (and, ideally, X+1) is defined:
 
-- [ ] src/common/ceph_releases.h (`ceph_release_t`)
```
Fixed again.

rzarzynski commented on src/test/cli/monmaptool/feature-set-unset-list.t:
```diff
@@ -41,23 +41,23 @@
-      optional:   [unknown(4096)]
-      required:   [kraken(1),octopus(32),unknown(4096)]
+      optional:   [umbrella(4096)]
+      required:   [kraken(1),octopus(32),umbrella(4096)]
```
Per the off-line discussion: 62fb916d730e5a815bac979408897f2407852dfa altered the value. The original intention was to use large number (like `4096`) and ensure it maps to `unknown`.

Need to choice different number.

rzarzynski commented on src/mon/MgrMonitor.cc:
```diff
@@ -97,6 +97,7 @@ static const std::map<uint32_t, std::set<std::string>>& always_on_modules() {
     { CEPH_RELEASE_REEF, octopus_modules },
     { CEPH_RELEASE_SQUID, octopus_modules },
     { CEPH_RELEASE_TENTACLE, octopus_modules },
+    { CEPH_RELEASE_UMBRELLA, octopus_modules },
```
Looks we're fine!

rzarzynski commented on src/osd/OSD.cc:
```diff
@@ -1766,7 +1766,8 @@ void OSDService::queue_for_snap_trim(PG *pg, uint64_t cost_per_object)
       /* We retain this legacy behavior for WeightedPriorityQueue.
-       * This branch should be removed after Squid.
+       * This branch should be removed after Umbrella (after consulting
+       * dev team)
```
@bill-scales points we should check also the counter itself and...

```
  sockeni01.front.sepia.ceph.com  rzarzynski   0c3131b  ~  dev  ceph-3  grep -r scrbcnt_write_blocked ./
./src/osd/osd_perf_counters.cc:  scrub_perf.add_u64_counter(scrbcnt_write_blocked, "write_blocked_by_scrub", "write blocked by scrub");
./src/osd/osd_perf_counters.h:  scrbcnt_write_blocked,
````

... the dropped block was the very sole user of it!

Let's drop the counter add well.

rzarzynski reviewed (commented):
2nd round.

aainscow committed:
osd: Remove unused osd_perf_counter "scrbcnt_write_blocked"

This counter is no longer needed in umbrella.

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

rzarzynski commented on src/osd/osd_perf_counters.h:
```diff
@@ -276,8 +276,6 @@ enum {
   /// # waiting on object events
   scrbcnt_blocked,
-  /// # write blocked by the scrub
-  scrbcnt_write_blocked,
```
👍

rzarzynski commented on doc/dev/release-checklists.rst:
```diff
@@ -108,15 +108,15 @@ Code cleanup
-- [X] remove qa/suites/upgrade/(X-3)-x-*
-- [X] update qa/suites/fs/upgrade/ to remove (X-3) and add (X-1); check with fs team to confirm / help
-- [X] update qa/ upgrade suites require-osd-release calls to tentacle
+- [x] create qa/workunits/test_telemetry_(X-1).sh
```
Huh, all this commit does is replacing big `X` with small `x`. Let's squash.

idryomov reviewed (approved):
Ack from the RBD perspective

vshankar reviewed (approved):
LGTM from cephfs pov.

batrick reviewed (changes requested):
Item:

Please make sure you've rebased recently.

```
- [ ] .github/workflows/redmine-upkeep.yml add release branch to pull_request_target trigger
```

needs addressed.

```
- [ ] src/script/ceph-release-notes (up to X)
```

status?

```
- [ ] look for features that (1) were present in X-2 and (2) have no
  client dependency and mark them `DEPRECATED` as of X.
```

I do not want to block Umbrella on this as it requires careful analysis. Please create an urgent RADOS tracker ticket for this to  be addressed for Vampire.

```
- [ ] qa/standalone/mon/misc adjust `TEST_mon_features` (add X cases and adjust `--mon-debug-no-require-X`)
```
status?

```
- [ ] OSDMap::get_min_compat_client: core team evaluate
```

@rzarzynski ?

```
- [ ] search code for `require_osd_release`
- [ ] search code for `min_mon_release`
```
?

(Let's chat in #ceph-upstream-releases-umbrella on ceph upstream slack for continued discussion as needed.)

**Nits**

* commit `ceph_release: update to tentacle` should refer to umbrella (possibly others too, reporting only once!)

aainscow committed:
script/ceph-release-notes: add umbrella

Just add umbrella.

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

aainscow committed:
doc: Check redmine-upkeep.yml to add umbrella.

Previous PR already added umbrella, therefore adding redmin-upkeep is a no-op.
Any future developer copying this needs to look at PR 67316

Signed-off-by: Alex Ainscow <aainscow@uk.ibm.com>

batrick reviewed (approved):
I'm good with merging this once the `make check` issues are sorted.

Matan-B commented on qa/suites/fs/upgrade/featureful_client/old_client/tasks/0-from/reef.yaml:
```diff
@@ -12,10 +12,8 @@ tasks:
       - cephadm
       - ceph-volume
-      - ceph-osd-classic
-      - ceph-osd-crimson
```
This isn’t a blocker, feel free to handle it however you prefer.
